### PR TITLE
Power generator fuel refactor

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Adds a player error message for when the generator already contains a sheet if a player tries to add one.

### EDIT:
Refactors fuel consumption by the power generators.

Removes SolidPlasma.cs, all fuel consumption will now be handled by PowerGenerator.cs
Fixes power generators burning plasma instantly after the first sheet of the stack gets removed.
Players will now be allowed to add stacks to the generator.
Adds a player message warning for when the player tries to turn on a generator without power.
A plasma sheet will no longer be required for maps, the bool startAsOn will add a the corresponding fuel of one sheet to the generator at the start.

### CODE REVIEW

**Comment by ReiderJack** [comment]
Does the generator code consume the item in the slot and leaves it as a null?

File: UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs [hunk 1]
```diff
@@ -154,8 +154,7 @@ public bool WillInteract(HandApply interaction, NetworkSide side)
 	{
 		if (!DefaultWillInteract.Default(interaction, side)) return false;
 		if (interaction.TargetObject != gameObject) return false;
-		if (interaction.HandObject != null &&
-				!Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma)) return false;
+		if (!Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma)) return false;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs]: Won't this mean you will not be able to toggle the power state of the generator?

File: UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs [hunk 2]
```diff
@@ -164,6 +163,11 @@ public void ServerPerformInteraction(HandApply interaction)
 	{
 		if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma))
 		{
+			if (itemSlot.Item)
+			{
+				Chat.AddWarningMsgFromServer(interaction.Performer, "The generator already contains a sheet!");
+				return;
+			}
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:133]: The generator uses ItemStorage now, so it should be able to contain up to fifty sheets.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:133]: it doesn't stack for some reason

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:133]: ye im making it stack ETA a couple hours

**Comment by Aranclanos** [comment]
What originally was a simple warning message pr ended up in a whole refactor of the generator, oh well

File: UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs [hunk 3]
```diff
@@ -31,9 +29,10 @@ public class PowerGenerator : NetworkBehaviour, ICheckedInteractable<HandApply>,
 	private ParticleSystem smokeParticles = default;
 
 	[SyncVar(hook = nameof(OnSyncState))]
-	private bool isOn = false;
-	private SolidPlasma burningSheet;
-	public bool IsFueled => itemSlot.IsOccupied || burningSheet != null;
+	private bool isOn;
+	private bool burningPlasma;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:33]: `burningPlasma` is never used.

File: UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs [hunk 4]
```diff
@@ -8,16 +8,14 @@ public class PowerGenerator : NetworkBehaviour, ICheckedInteractable<HandApply>,
 {
 	[Tooltip("Whether this generator should start running when spawned.")]
 	[SerializeField]
-	private bool startAsOn = false;
+	private bool startAsOn;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:11]: Produces a compiler warning. https://i.imgur.com/4YWm1cO.png.

**Comment by NoooneyDude** [comment]
Couldn't start a generator after refueling it from empty, either using the starting generator, a spare generator, the new or old plasma sheets. This is while connected to a headless server.

plasma generator

**Comment by ReiderJack** [comment]
Why plasma sheet prefabs are deleted?

File: UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs (hunk 4 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:11]: yeah, you need to give it a default value

**Pull request closed without merging**